src/template-linter.ts

```diff
@@ -12,6 +12,7 @@ import Server from './server';
 import { Project } from './project';
 import { getRequireSupport } from './utils/layout-helpers';
 import { getFileRanges, RangeWalker } from './utils/glimmer-script';
+import semver, { SemVer } from 'semver';
```

Comment:
nope, it's extra bundle size

Reply:
@lifeart why not? The raw string `^5.13.0' for example can't be read as `versionString[0]` otherwise you get a version `^` no?

Reply:
@arthur5005 it's not a case, we already cleaning value here: https://github.com/ember-tooling/ember-language-server/blob/master/src/utils/layout-helpers.ts#L293

Reply:
@lifeart just tested how `clean` works, I think it doesn't work how we think it's supposed to work, unless I'm missing something.
<img width="906" alt="Screenshot 2024-11-22 at 8 50 24 AM" src="https://github.com/user-attachments/assets/782eba51-ef69-4b07-bbc3-505901a85f05">

Reply:
@arthur5005  oh my, thank you for pointing to it!

Reply:
^- @NullVoxPopuli To @lifeart's point, I think how we parse dependency versions is the real problem here, I'm imagining the version on the dep meta is null for all unpinned projects. :\